service/service.go

```diff
@@ -13,10 +13,10 @@ import (
 type Service struct {
 	repo       repository
 	privateKey *rsa.PrivateKey
-	provider
+	prov       contextValueProvider
```

Comment:
please don't call variables as `pov`, because reading code below like `s.prov.GetCurrentUser` is not fully clear where do you call `GetCurrentUser`.